### DISCUSSION TITLE
Cyberghost - Can´t get it running

### DISCUSSION THREAD

**Discussion by fluppi82** [qdm12/gluetun#480] [Q&A]
Can please someone tell me what the CLI command is for Cyberghost I have to use in Putty. I tried everything and can´t get it running. Now I get the error message: 

```
ERROR cannot read secret file: read /gluetun/client.key: is a directory
```

I´m in total despair and have no more idea. Please help.

One of the few things I tried:

```
docker run -d --name gluetun --cap-add=NET_ADMIN \
-e VPNSP="cyberghost" -e REGION="Italy" \
-e OPENVPN_USER=ABC -e OPENVPN_PASSWORD=123 \
-v /yourpath/client.key:/gluetun/client.key:ro \
-v /yourpath/client.crt:/gluetun/client.crt:ro \
qmcgaw/gluetun
```

```
docker run -d --name gluetun --cap-add=NET_ADMIN \
-e VPNSP="cyberghost" -e REGION="Italy" \
-e OPENVPN_USER=ABC -e OPENVPN_PASSWORD=123 \
-v /yourpath:/gluetun \
qmcgaw/gluetun
```

**Comment by qdm12** [comment]
What's the Docker host operating system?

Bind mounting the file directly works well on Linux hosts, but not so much on Windows and OSX hosts.
The second command you have where you bind mount the entire directory should work on all platforms though. Do you get the same error `ERROR cannot read secret file: read /gluetun/client.key: is a directory` with that second command?

Have a look at `/yourpath` in your host, maybe client.key is a directory created when you ran the first command. In this case remove the directory, add your client.key file and then try again with your second command.

I'll try to answer quickly, please don't despair we'll get this sorted out 😉

**Comment by fluppi82** [comment]
System is Raspberry Pi 4b + Openmediavault

Everything else I will answer tomorrow. Need some sleep so I can think better. But thank you so far. 👍

**Comment by fluppi82** [comment]
I deleted the container and /yourpath. So I basically started over with the command:

```sh
docker run -d --name gluetun --cap-add=NET_ADMIN \
-e VPNSP="cyberghost" -e REGION="Italy" \
-e OPENVPN_USER=ABC -e OPENVPN_PASSWORD=123 \
-v /yourpath:/gluetun \
qmcgaw/gluetun
```

Now everything is working. I guess the other command messed something up. 

I still get three warnings:

```
2021/06/08 17:10:26 WARN openvpn: 'link-mtu' is used inconsistently, local='link-mtu 1569', remote='link-mtu 1549',
2021/06/08 17:10:26 WARN openvpn: 'auth' is used inconsistently, local='auth SHA256', remote='auth [null-digest]',
2021/06/08 17:10:26 WARN openvpn: 'keysize' is used inconsistently, local='keysize 256', remote='keysize 128',
```

Should I fix that and how?

**Reply by qdm12** [reply]
That's just their VPN server pushing different options that they have on their client configurations. It's fine as openvpn will use the options pushed from the server.

Although could you download any openvpn config file from Cyberghost and check if their `auth sha256` line is still there? How about their `cipher` / `data-cipher` line(s), are they still set to `aes-256-cbc`?

For the link-mtu thing you can set the MSS fix environnement variable (check the gluetun wiki and openvpn's doc, I'm not that familiar with it sorry)
